platforms/core-configuration/declarative-dsl-tooling-builders/src/main/kotlin/org/gradle/declarative/dsl/tooling/builders/DeclarativeSchemaModelBuilder.kt

```diff
@@ -19,17 +19,28 @@ package org.gradle.declarative.dsl.tooling.builders
 import org.gradle.api.Project
 import org.gradle.declarative.dsl.schema.AnalysisSchema
 import org.gradle.declarative.dsl.tooling.models.DeclarativeSchemaModel
-import org.gradle.internal.declarativedsl.evaluator.DeclarativeSchemaRegistry
+import org.gradle.internal.build.BuildState
+import org.gradle.internal.declarativedsl.project.projectEvaluationSchema
+import org.gradle.plugin.software.internal.SoftwareTypeRegistry
 import org.gradle.tooling.provider.model.ToolingModelBuilder
+import org.gradle.tooling.provider.model.internal.BuildScopeModelBuilder
 import java.io.Serializable
 
 
-class DeclarativeSchemaModelBuilder(private val registry: DeclarativeSchemaRegistry) : ToolingModelBuilder {
+class DeclarativeSchemaModelBuilder(private val softwareTypeRegistry: SoftwareTypeRegistry) : ToolingModelBuilder, BuildScopeModelBuilder {
+
+    override fun create(target: BuildState?): Any {
+        val schema = projectEvaluationSchema(softwareTypeRegistry).analysisSchema
+        return DefaultDeclarativeSchemaModel(schema)
+    }
+
     override fun canBuild(modelName: String): Boolean =
         modelName == "org.gradle.declarative.dsl.tooling.models.DeclarativeSchemaModel"
 
-    override fun buildAll(modelName: String, project: Project): Any =
-        DefaultDeclarativeSchemaModel(registry.projectSchema())
+    override fun buildAll(modelName: String, project: Project): Any {
+        // DefaultDeclarativeSchemaModel(registry.projectSchema())
```

Comment:
❌ This commented statement should be removed

Reply:
Fixed.